### PR TITLE
Add SIP for distributing Spin applications using OCI registries

### PR DESCRIPTION
**Pull request by radu-matei**
This extracts just the improvement proposal from #1014, so we can start iterating on the implementation separately.

Signed-off-by: Radu Matei <radu.matei@fermyon.com>

### CODE REVIEW

File: docs/content/sips/008-using-oci-registries.md [hunk 1]
```diff
@@ -0,0 +1,364 @@
...
+The goal is to have at least one minor release of Spin with `spin oci` before
+updating the *default* `spin deploy` behavior to expect an OCI registry.
+Once Fermyon Platform or Fermyon Cloud support accepting OCI references,
+the intention is to implement `spin deploy --oci` functionality before
+updating the default behavior.
+
+### Implementation status
+
+The `spin oci push`, `spin oci pull`, and `spin oci run` commands are currently
+implemented in this prototype. The implementation uses the
```
Comment (itowlson) [inline docs/content/sips/008-using-oci-registries.md:346]: Ideally link the PR that contains the prototype since it's no longer part of this one.

**Review by itowlson — APPROVED** [review]
LGTM, one minor suggestion related to the separation of this PR

File: docs/content/sips/008-using-oci-registries.md [hunk 2]
```diff
@@ -0,0 +1,364 @@
+title = "SIP 008 - Distributing Spin applications using OCI registries"
+template = "main"
+date = "2023-01-04T01:01:01Z"
```
Comment (kate-goldenring) [inline docs/content/sips/008-using-oci-registries.md:3]: Is it worth bumping the date?

**Review by fibonacci1729 — APPROVED** [review]

File: docs/content/sips/008-using-oci-registries.md (hunk 2 above)
Comment (radu-matei) [reply docs/content/sips/008-using-oci-registries.md:3]: I do like to have a record of when this was originally written first.

File: docs/content/sips/008-using-oci-registries.md (hunk 1 above)
Comment (radu-matei) [reply docs/content/sips/008-using-oci-registries.md]: Good catch, updated.

**Comment by radu-matei** [comment]
Added a link to the prototype in #1014 and updated the SIP index.

**Review by michelleN — COMMENTED** [review]
👍

**Pull request merged**